File: admin/client/tools/interpreter.ts
import { z } from "zod";

export const E2BInterpreterToolConfig = z.object({
  name: z.literal("interpreter"),
  label: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  enabled: z.boolean().nullable().optional(),
  config: z
    .object({
      api_key: z
        .string()
        .nullable()
        .refine(
          (data) => {
            return data && data.length > 0;
          },
          { message: "API Key is required" },
        ),
    })
    .nullable()
    .optional(),
});
export type E2BInterpreterToolConfigType = z.infer<
  typeof E2BInterpreterToolConfig
>;
export const DEFAULT_E2B_INTERPRETER_TOOL_CONFIG = {
  label: "Code Interpreter",
  description: "",
  config: {
    api_key: "",
  },
  enabled: false,
};

File: src/models/tools/interpreter.py
from typing import Dict, ClassVar, Literal
from pydantic import BaseModel, Field


class E2BInterpreterToolConfig(BaseModel):
    api_key: str | None = Field(
        default=None,
        description="The API key to use for the E2B Interpreter.",
    )


class E2BInterpreterTool(BaseModel):
    config_id: ClassVar[str] = "interpreter"
    name: Literal["interpreter"] = "interpreter"
    tool_type: Literal["local"] = "local"
    label: Literal["E2B Interpreter"] = "E2B Interpreter"
    description: str = "Execute Python code in a sandbox environment."
    config: E2BInterpreterToolConfig | None = Field(
        default=E2BInterpreterToolConfig(),
    )
    enabled: bool = False
